src/client.rs

```diff
@@ -125,6 +125,26 @@ impl<'a> Client<'a> {
         .into_index(self))
     }
 
+    /// Delete an index from its UID if it exists.
+    /// To delete an index if it exists from the [index object](../indexes/struct.Index.html), use
+    /// [the delete_if_exists method](../indexes/struct.Index.html#method.delete_if_exists).
+    pub async fn delete_index_if_exists(&self, uid: &str) -> Result<bool, Error> {
+        match self.delete_index(uid).await {
+            Ok (_) => return Ok(true),
+            Err (error) => {
+                match error {
+                    Error::MeiliSearchError {
+                        message: _,
+                        error_code: ErrorCode::IndexNotFound,
+                        error_type: _,
+                        error_link: _,
+                    } => return Ok(false),
+                    _ => return Err(error),
+                };
+            },
```

Comment:
Is clippy ok with this? It seems that `if let` should be used instead of `match` but it would be even better to match the inner pattern directly in the first match. I did not test the suggested code.
```suggestion
            Err (Error::MeiliSearchError {
                message: _,
                error_code: ErrorCode::IndexNotFound,
                error_type: _,
                error_link: _,
            }) => Ok(false),
            Err(error) => Err(error),
```

Reply:
Clippy gave no errors here, but I also don't mind changing it. I will try out your updated.

Reply:
The compiler complained about no return statements, but other than that this change worked. I added the returns and then made this same change in both places.

Reply:
Yeah that's because there is a semicolon after the match. Try removing the semicolon and you will be able to remove the returns.

Reply:
Yep, I should have noticed that...it's been a long week 😄 . Returns are removed now.